### PR TITLE
Bump node from 12 to 16

### PR DESCRIPTION
**Pull request by quinnjn**
###  Summary

Bumps node12 to node16. Also suggesting at least node 16 for install. Node 12 is EOL.

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: package.json
Comment (mwbrooks) [inline package.json:29]: Nit and may not matter: It looks like `node@16` ships with `npm@7.10.0`
image
Reference: https://nodejs.org/en/download/releases/

**Review by mwbrooks — APPROVED** [review]
Thanks @quinnjn! We were just discussing bumping all of the Slack node projects from 12 to 16, so this PR is greatly appreciated!

I've left a small nit/question but it looks good to me. I've added a few other maintainer's who are active on this project to approve the PR before we move forward.

**Review by srajiang — APPROVED** [review]
Yep, looks good - would be good to make parallel changes in node-slack-sdk and bolt-js as 12 is EOL.

**Review by filmaj — APPROVED** [review]
LGTM as long as the npm minimum version can match the version that ships with node 16.

File: package.json
Comment (quinnjn) [reply package.json]: Ah good shout. Fixed!

**Comment by mwbrooks** [comment]
Beauty @quinnjn 👌🏻 Thanks for the changes and you've got ✅ ✅ ✅ all around! I've merged the pull request and we'll put together a release soon!

**Pull request merged**
